### PR TITLE
docs: add vs code extensions documentation

### PR DESCRIPTION
**Pull request by ericpaulsen**
this PR ports over the VS Code extensions doc we wrote for v1. i've included updated semantics, screenshots, and code blocks.

### CODE REVIEW

File: docs/ides/vscode-extensions.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+## Using the public extensions marketplaces
+
+You can manually add an extension while you're working in the Code Web IDE. The
+extensions can be from Coder's public marketplace, Eclipse Open VSX's public
+marketplace, or the Eclipse Open VSX _local_ marketplace.
+
+![Code Web Extensions](../images/ides/code-web-extensions.png)
+
+> Note: code-server cannot legally connect to Microsoft's public
```
Comment (matifali) [inline docs/ides/vscode-extensions.md:26]: Maybe reword is that Microsoft doesn't allow connecting any other vscode based idea to use their marketplace.

Comment (ericpaulsen) [reply docs/ides/vscode-extensions.md]: updated

**Comment by ericpaulsen** [comment]
@matifali i'm running `make fmt` with my commits, but the check is still failing. any idea why?

**Comment by matifali** [comment]
@ericpaulsen `fmt` is now passing. But as I have pushed to the repo. Let's wait for @stirby to review.

File: docs/ides/vscode-extensions.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+For your local VS Code to pickup extension files in your Coder workspace,
+include this command in your `startup_script`, or run in manually in your
+workspace terminal:
+
+```console
+code --extensions-dir ~/.vscode-server/extensions --install-extension "$extension"
+```
+
+![Microsoft extension marketplace](../assets/workspaces/local-vs-code-marketplace.png)
```
Comment (matifali) [inline docs/ides/vscode-extensions.md:133]: @ericpaulsen This screenshot doesn't seem to exist.

Comment (ericpaulsen) [reply docs/ides/vscode-extensions.md]: removed - thank you.

**Review by matifali — APPROVED** [review]
